code/modules/events/antagonist_critter_ghost_respawn.dm

```diff
@@ -20,35 +21,33 @@
 		if (..())
 			return
 
-		var/input = alert(usr, "Choose the critter type?", src.name, "Random", "Custom")
-		if (!input)
-			return
-		else if (input == "Custom")
-			input = input("Enter a /mob/living/critter path", src.name, null) as null|text
-			input = get_one_match(input, "/mob/living/critter")
-		else //random
-			input = null
+		switch (alert(usr, "Choose the critter type?", src.name, "Random", "Custom"))
+			if ("Custom")
+				src.critter_type = input("Enter a /mob/living/critter path", src.name, null) as null|text
```

Comment:
I feel like "Enter a /mob/living/critter path" is kinda misleading because they don't have to type `/mob/living/critter/aquatic/fish/damsel` but can just type `damsel`, right?